### PR TITLE
use suggestions for "enum instead of variant" error

### PR DESCRIPTION
**Pull request by euclio**

### CODE REVIEW

File: src/librustc_resolve/error_reporting.rs [hunk 1]
```diff
@@ -293,13 +293,20 @@ impl<'a> Resolver<'a> {
...
-                    err.note(&format!("did you mean to use one \
-                                       of the following variants?\n{}",
-                        variants.iter()
-                            .map(|suggestion| path_names_to_string(suggestion))
-                            .map(|suggestion| format!("- `{}`", suggestion))
-                            .collect::<Vec<_>>()
-                            .join("\n")));
+                    if !variants.is_empty() {
+                        let msg = if variants.len() == 1 {
+                            "use the enum's variant"
```
Comment (varkor) [inline src/librustc_resolve/error_reporting.rs]:
```suggestion
                            "try using the enum's variant"
```
(For consistency with existing suggestions.)

File: src/librustc_resolve/error_reporting.rs [hunk 2]
```diff
@@ -293,13 +293,20 @@ impl<'a> Resolver<'a> {
...
+                        } else {
+                            "use one of the enum's variants"
```
Comment (varkor) [inline src/librustc_resolve/error_reporting.rs]:
```suggestion
                            "try using one of the enum's variants"
```

**Comment by varkor** [comment]
r=me with the minor phrasing change.

**Comment by euclio** [comment]
@varkor Fixed.

**Pull request merged**